### PR TITLE
Added: Basic Page products, login, cart

### PR DESCRIPTION
**Pull request by hussiantabha**
Added the basic structure of the page. Functionalities it not working as of now.

### CODE REVIEW

File: src/Components/Nav.js [hunk 1]
```diff
@@ -18,15 +18,21 @@ const NavBar = () => {
...
-          <div class="badges">
-            <i class="fas fa-shopping-cart nav-cart"></i>
-            <span class="badge">2</span>
-          </div>
+          <Link to="/login">
+            <button className="btn btn-primary-outline"> Login </button>
+          </Link>
+          <Link to="/wishlist">
+            <div class="badges">
+              <i class="fas fa-heart nav-cart"></i>
```
Comment (Bhakti199) [inline src/Components/Nav.js]: you may use react-icons

File: src/Components/listingProducts.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                <input type="checkbox" />
+                <span>3 star & above</span>
+              </div>
+              <div>
+                <input type="checkbox" />
+                <span>2 star & above</span>
+              </div>
+              <div>
+                <input type="checkbox" />
+                <span>1 star & above</span>
```
Comment (Bhakti199) [inline src/Components/listingProducts.js]: Instead here you can map over `[1,2,3,4]`

File: src/Components/listingProducts.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+            </div>
+            <div class="filter-category">
+              <h3>Sort By</h3>
+              <div>
+                <input type="radio" name="value" />
+                <span>Price - Low to High</span>
+              </div>
+              <div>
+                <input type="radio" name="value" />
+                <span>Price - High to Low</span>
```
Comment (Bhakti199) [inline src/Components/listingProducts.js]: use `label` here

File: src/Components/listingProducts.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+                <span>Day To Day Tea</span>
+              </div>
+              <div>
+                <input type="checkbox" />
+                <span>Exotic Tea</span>
+              </div>
+              <div>
+                <input type="checkbox" />
+                <span>Premium Tea</span>
+              </div>
```
Comment (Bhakti199) [inline src/Components/listingProducts.js]: use `className` not `class`

File: src/Pages/Cart.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                src="../assets/exotic.png"
+                class="cart-card-img"
+                alt="Exotica Tea"
+              />
+            </div>
+            <div class="cart-card-content">
+              <h2>Exotic Tea</h2>
+              <div class="card-price">
+                <span class="price-scratch">₹500/-</span>
+                <span>₹299/-</span>
```
Comment (Bhakti199) [inline src/Pages/Cart.js]: use `className` not `class`

File: src/Pages/Login.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { NavBar } from "../Components/Nav";
```
Comment (Bhakti199) [inline src/Pages/Login.js:3]: for importing components use may use single index file to export all componnts from it.

File: src/index.js [hunk 7]
```diff
@@ -5,6 +5,10 @@ import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter, Route, Routes } from "react-router-dom";
 import { Products } from "./Pages/Products";
+import Wishlist from "./Pages/Wishlist";
+import Cart from "./Pages/Cart";
+import Login from "./Pages/Login";
+import SignUp from "./Pages/SignUp";
```
Comment (Bhakti199) [inline src/index.js:11]: same here as above , export it from index file

**Review by Bhakti199 — COMMENTED** [review]
Replace all `class` with `className`.
Also I suggested some changes below.

**Review by JantuDeb — APPROVED** [review]
As @Bhakti199  suggested replace `class` with `className`.
Create reusable components for product cards, load data from API then use it in UI.

File: src/App.js [hunk 8]
```diff
@@ -1,20 +1,16 @@
...
-  //   const rew = fetch("/api/products")
-  //     .then((res) => res.json())
-  //     .then((data) => setData(data.products));
-  // };
-  // useEffect(getData, []);
+function App() {
+  const { products } = useContext(FilterContext);
+  console.log(products);
   return (
     <>
```
Comment (JantuDeb) [inline src/App.js]: You can remove this empty react fragment `<> </>`

File: src/Components/listingProducts.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
+import React, { useContext } from "react";
+import { FilterContext } from "../Context";
+import "../App.css";
+
+const ListingProducts = () => {
+  const { products } = useContext(FilterContext);
+  console.log(products);
```
Comment (JantuDeb) [inline src/Components/listingProducts.js]: remove logs from deploy build

File: src/Components/listingProducts.js [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+                  Add to Cart
+                </a>
+                <a href="./cart.html" className="btn btn-primary">
+                  Buy Now
+                </a>
+              </div>
+              <button className="btn-primary card-like-btn">
+                <i className="fas fa-heart"></i>
+              </button>
+            </div> */}
```
Comment (JantuDeb) [inline src/Components/listingProducts.js]: remove commented codes

File: src/Components/listingProducts.js [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+                  <div className="card-btn-container">
+                    <button className="btn btn-primary-outline">
+                      Add to Cart
+                    </button>
+                    <button className="btn btn-primary">Buy Now</button>
+                  </div>
+                  <button className="btn-primary card-like-btn">
+                    <i className="fas fa-heart"></i>
+                  </button>
+                </div>
```
Comment (JantuDeb) [inline src/Components/listingProducts.js]: You can extract these codes to a separate component. You will be able to use it for wishlist or cart also

File: src/Context.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import React, { createContext, useState, useEffect } from "react";
+import axios from "axios";
+
+const FilterContext = createContext({});
+
+const FilterContextProvider = ({ children }) => {
+  const [products, setProducts] = useState([]);
+  const getData = async () => {
+    const data = await axios.get("api/products");
+    setProducts(data.data.products);
```
Comment (JantuDeb) [inline src/Context.js]: call asynchronous code in `try{}catch(error){}` block

**Review by JantuDeb — APPROVED** [review]
Suggested a changes

**Pull request merged**